### PR TITLE
Support CREATE2 opcode in Constantinople

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

**Comment by gumb0** [comment]
Still need to add it to instructions lib.

File: include/evmc/evmc.h [hunk 1]
```diff
@@ -70,7 +70,8 @@ enum evmc_call_kind
     EVMC_CALL = 0,         /**< Request CALL. */
     EVMC_DELEGATECALL = 1, /**< Request DELEGATECALL. The value param ignored. */
     EVMC_CALLCODE = 2,     /**< Request CALLCODE. */
-    EVMC_CREATE = 3        /**< Request CREATE. Semantic of some params changes. */
+    EVMC_CREATE = 3,       /**< Request CREATE. Semantic of some params changes. */
```
Comment (chfast) [inline include/evmc/evmc.h]: You can remove the "Semantic of some params changes" part in both cases.

Comment (axic) [reply include/evmc/evmc.h]: I think these should state which `evmc_revision` they are valid in (frontier, homestead, byzantium, constantinople).

File: include/evmc/evmc.h [hunk 2]
```diff
@@ -116,6 +118,13 @@ struct evmc_message
      */
     struct evmc_uint256be code_hash;
 
+    /**
+     * The optional value used in new contract address construction.
+     *
+     *  Ignored unless kind == EVMC_CREATE2.
+     */
+    struct evmc_uint256be salt;
```
Comment (chfast) [inline include/evmc/evmc.h]: How about we name it `create_salt`?

File: include/evmc/evmc.h [hunk 3]
```diff
@@ -117,6 +118,13 @@ struct evmc_message
      */
     struct evmc_uint256be code_hash;
 
+    /**
+     * The optional value used in new contract address construction.
+     *
+     *  Ignored unless kind == EVMC_CREATE2.
```
Comment (axic) [inline include/evmc/evmc.h]: `unless kind is EVMC_CREATE2`.

File: test/vmtester/tests.cpp [hunk 4]
```diff
@@ -12,7 +12,6 @@ static_assert(sizeof(evmc_uint256be) == 32, "evmc_uint256be is too big");
 static_assert(sizeof(evmc_address) == 20, "evmc_address is too big");
 static_assert(sizeof(evmc_result) <= 64, "evmc_result does not fit cache line");
 static_assert(sizeof(evmc_instance) <= 64, "evmc_instance does not fit cache line");
-static_assert(sizeof(evmc_message) <= 18 * 8, "evmc_message not optimally packed");
```
Comment (axic) [inline test/vmtester/tests.cpp]: Should this be kept with an updated size?

Comment (chfast) [reply test/vmtester/tests.cpp]: It does not anything useful. I was just checking obvious struct fields alignment.

File: include/evmc/evmc.h (hunk 2 above)
Comment (axic) [reply include/evmc/evmc.h]: `create2_salt` for explicitlness?

Comment (axic) [reply include/evmc/evmc.h]: Otherwise `creation_salt`?

Comment (chfast) [reply include/evmc/evmc.h]: `create2_salt` :+1: 
Similarly to `create_address`.

**Comment by gumb0** [comment]
Renamed to `create_salt`

**Comment by axic** [comment]
This also needs an ABI version bump.

**Comment by gumb0** [comment]
Bumped ABI version to 2

**Pull request merged**